src/Provider/Provider.Web/Services/VacancyLocationService.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Esfa.Recruit.Shared.Web.Extensions;
+using Esfa.Recruit.Vacancies.Client.Application.Validation;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Client;
+
+namespace Esfa.Recruit.Provider.Web.Services;
+
+public interface IVacancyLocationService
+{
+    public Task<List<Address>> GetVacancyLocations(Vacancy vacancy);
+    public Task<UpdateVacancyLocationsResult> UpdateDraftVacancyLocations(Vacancy vacancy, VacancyUser user, AvailableWhere availableWhere, List<Address> locations = null, string locationInformation = null);
+}
+
+public record UpdateVacancyLocationsResult(EntityValidationResult ValidationResult);
+
+public class VacancyLocationService(IRecruitVacancyClient recruitVacancyClient, IEmployerVacancyClient employerVacancyClient): IVacancyLocationService
+{
+    public async Task<List<Address>> GetVacancyLocations(Vacancy vacancy)
+    {
+        ArgumentNullException.ThrowIfNull(vacancy);
+        var employerProfile = await recruitVacancyClient.GetEmployerProfileAsync(vacancy.EmployerAccountId, vacancy.AccountLegalEntityPublicHashedId);
+
+        var employerData = await employerVacancyClient.GetEditVacancyInfoAsync(employerProfile.EmployerAccountId);
+        var legalEntity = employerData.LegalEntities.First(l => l.AccountLegalEntityPublicHashedId == employerProfile.AccountLegalEntityPublicHashedId);
```

Comment:
i think i would be tempted to change this to FirstOrDefault - only to cater for scenarios where the AccountLegalEntity is removed from the account by the employer.